docs/use_cases/advanced_retrieval_augmented_generation.md

```diff
@@ -104,8 +115,10 @@ document_store = InMemoryDocumentStore()
 writer = DocumentWriter(document_store=document_store)
 ```
 
+Indexing organizes the pre-processed data in a structured format that can be quickly and accurately retrieved by the RAG system. This step involves creating an optimized database of tagged and categorized chunks.

```

Comment:
@xaiguy 
In this section called "Indexing" there is code for creating the database, but not for indexing itself?

Reply:
That's because everything is done automatically based on the data we're feeding into the pipeline. I agree that I should add sentence or two about what's going on there since there is no seperate indexing component in Haystack. It basically depends on how the data was pre-processed (embeddings, chunks, column names etc.).

Reply:
can you add your sentence or two here, and I'll integrate it into the article text?